### PR TITLE
Change profile view's display name and detailed status's timestamp to link to canonical URL

### PR DESCRIPTION
**Pull request by trwnh**
Follow-up to #20540 -- since these links are in essence "terminal", it can be argued that users may not expect them to open a new tab on their local server, but instead to open the canonical URL.

### CODE REVIEW

File: app/javascript/mastodon/features/account/components/header.js [hunk 1]
```diff
@@ -328,12 +328,12 @@ class Header extends ImmutablePureComponent {
...
           </div>
 
-          <div className='account__header__tabs__name'>
+          <a className='account__header__tabs__name' href={`${account.get('url')}`} target='_blank' rel='noopener noreferrer'>
             <h1>
               <span dangerouslySetInnerHTML={displayNameHtml} /> {badge}
               <small>@{acct} {lockedIcon}</small>
             </h1>
-          </div>
+          </a>
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/account/components/header.js:336]: I don't know if I'd have that be a link.

File: app/javascript/mastodon/features/status/components/detailed_status.js [hunk 2]
```diff
@@ -258,6 +259,15 @@ class DetailedStatus extends ImmutablePureComponent {
       );
     }
 
+    if (status.getIn(['account', 'username']) !== status.getIn(['account', 'acct'])) {
+      externalLink = (
+        <React.Fragment>
+          <React.Fragment> </React.Fragment>
+          <Icon id='external-link' />
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/status/components/detailed_status.js:266]: Maybe a `title` and/or `aria-label`?

File: app/javascript/mastodon/features/account/components/header.js (hunk 1 above)
Comment (trwnh) [reply app/javascript/mastodon/features/account/components/header.js:336]: The reasoning behind this was to give users an easy way to open the remote profile without menu diving, but I'm open to alternative approaches too

File: app/javascript/mastodon/features/status/components/detailed_status.js (hunk 2 above)
Comment (trwnh) [reply app/javascript/mastodon/features/status/components/detailed_status.js:266]: What would they contain?

Comment (ClearlyClaire) [reply app/javascript/mastodon/features/status/components/detailed_status.js:266]: Something like “External link”, “External source” or “Original source”?

Comment (trwnh) [reply app/javascript/mastodon/features/status/components/detailed_status.js]: Added, in what I assume is the correct way but idk really

File: app/javascript/mastodon/features/status/components/detailed_status.js [hunk 3]
```diff
@@ -276,8 +288,9 @@ class DetailedStatus extends ImmutablePureComponent {
           {media}
 
           <div className='detailed-status__meta'>
-            <a className='detailed-status__datetime' href={`/@${status.getIn(['account', 'acct'])}\/${status.get('id')}`} target='_blank' rel='noopener noreferrer'>
+            <a className='detailed-status__datetime' href={`${status.get('url')}`} target='_blank' rel='noopener noreferrer' title='${intl.formatMessage(messages.canonical_link)}' aria-label='${intl.formatMessage(messages.canonical_link)}'>
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/status/components/detailed_status.js:291]: Hmmm I'm not too sure about adding a label/title here.

Comment (trwnh) [reply app/javascript/mastodon/features/status/components/detailed_status.js:291]: My thinking is that you can hover over the timestamp, so you should be able to see a hint about what it leads to? The external icon is just part of the link at the end

Comment (ClearlyClaire) [reply app/javascript/mastodon/features/status/components/detailed_status.js:291]: I really don't think `aria-label` is appropriate here, the thing's primary purpose is that it's a timestamp, it's not a visual-only thing either, I think `aria-label` would incorrectly obscure the timestamp.

Comment (trwnh) [reply app/javascript/mastodon/features/status/components/detailed_status.js:291]: Would a `title` but no `aria-label` be appropriate, or should there be a different `aria-label`?

**Comment by mjankowski** [comment]
The relevant area here has moved to a new file (header.jsx), and the area has changed a bit (now includes domain pill) since the original PR ... it also seems like the conversation sort of fizzled out.

Do you want to rebase and re-review this in the new file location, or should we close?

**Comment by mjankowski** [comment]
Closing on assumption that the updated view/js here may have made this not relevant, and due to no rebase in ~18 months.

If this is still relevant, can you rebase/reopen?

**Pull request closed without merging**